src/org/folio/rest_v2/eureka/Eureka.groovy

```diff
@@ -212,4 +215,139 @@ class Eureka extends Base {
 
     return this
   }
+
+  /**
+   * Get Configured Tenants on Environment Namespace.
+   * @return Map of EurekaTenant objects.
+   */
+  Map<String, EurekaTenant> getExistedTenantsFlow() {
+    /** Configured Tenants in Environment (namespace) */
+    Map<String, EurekaTenant> configuredTenantsMap
+
+    // Get configured Tenants from the Environment (namespace)
+    configuredTenantsMap = Tenants.get(kong).getTenants().collectEntries { tenant -> [tenant.tenantName, tenant] }
+
+    // Get enabled (entitled) applications for configured Tenants
+    configuredTenantsMap.each { tenantName, tenant ->
+      /** Enabled (entitled) applications for Tenant*/
+      Map<String, String> enabledAppsMap = [:]
+
+      // Get enabled applications from the Environment
+      Tenants.get(kong).getEnabledApplications(tenant).each { appId, app ->
+        enabledAppsMap.put(appId.split("-\\d+\\.\\d+\\.\\d+")[0], appId)
+      }
+
+      // Assign enabled applications to Tenant object
+      tenant.applications = enabledAppsMap.clone() as Map
+    }
+
+    return configuredTenantsMap
+  }
+
+  /**
+   * Get Existed Applications on Environment Namespace.
+   * @return Map of Application Name and Application ID.
+   */
+  static Map<String, String> getEnabledApplicationsFlow(Map<String, EurekaTenant> tenants) {
+    /** Enabled Applications in Environment */
+    Map <String, String> enabledAppsMap = [:]
+
+    // Get enabled applications from EurekaTenant List of objects
+    tenants.values().each {tenant ->
+      tenant.applications.each {appName, appId ->
+        enabledAppsMap.put(appName, appId)
+      }
+    }
+
+    return enabledAppsMap
+  }
+
+  /**
+   * Update Application Descriptor Flow.
+   * @param applications Map of enabled applications in namespace.
+   * @param module FolioModule object.
+   * @return Map of EurekaTenant objects.
+   */
+  Eureka updateAppDescriptorFlow(Map<String, String> applications, FolioModule module) {
+    /** Enabled Application Descriptors Map */
+    Map<String, Object> appDescriptorsMap = [:]
+
+    //Get application descriptors for enabled applications in namespace
+    applications.each { appName, appId ->
+      def appDescriptor = Applications.get(kong).getRegisteredApplication(appId)
+      if (appDescriptor['modules'].any { it['name'] == module.name }) {
+        appDescriptorsMap.put(appId, appDescriptor)
+      }
+    }
+
+    // Get Application Descriptor Updated with New Module Version
+    appDescriptorsMap.each { appId, descriptor->
+      // Get Incremental Number for New Application Version
+      String incrementalNumber = descriptor['version'].toString().tokenize('.').last().toInteger() + 1
+
+      // Update existing Application Descriptor with New Module Version
+      Map updatedAppDescriptor = getUpdatedApplicationDescriptor(descriptor as Map, module, incrementalNumber)
+
+      // Register Updated Application Descriptor to Environment
+      // TODO: uncomment following line after testing
```

Comment:
Is testing complete?

Reply:
**folio-eureka-volya** environment is a bit compromised.
So there is deploed very fresh **folio-tmp-etest01** environment.
Today this line will be re-tested in scope of end to end pipeline testing.